sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -0,0 +1,546 @@
+     * {@link SignatureAlgorithm#RS512 RS512}, {@link SignatureAlgorithm#RS256 RS256}, {@link SignatureAlgorithm#RS384 RS384} and
+     * {@link SignatureAlgorithm#RS512 RS512}</p>
+     *
+     * @param algorithm The algorithm to use for signing.
+     * @param data The raw content against which signature is to be verified.
+     * @param signature The signature to be verified.
+     * @throws ResourceNotFoundException if the key cannot be found for verifying.
+     * @return The {@link Boolean} indicating the signature verification result.
+     */
+    public Mono<VerifyResult> verifyData(SignatureAlgorithm algorithm, byte[] data, byte[] signature) {
+        return withContext(context -> verifyData(algorithm, data, signature, context));
+    }
+
+    Mono<VerifyResult> verifyData(SignatureAlgorithm algorithm, byte[] data, byte[] signature, Context context) {
+        Objects.requireNonNull(algorithm);
+        boolean keyAvailableLocally = ensureValidKeyAvailable();
+
+        if(!keyAvailableLocally) {
+            return cryptographyServiceClient.verifyData(algorithm, data, signature, context);
+        }
+
+        if (!checkKeyPermissions(this.key.keyOps(), KeyOperation.VERIFY)){
+            return Mono.error(new UnsupportedOperationException(String.format("Verify Operation is not allowed for key with id %s", this.key.kid())));
+        }
+
+        switch(this.key.kty()){
+            case RSA:
+            case RSA_HSM:
+                return rsaKeyCryptographyClient.verifyDataAsync(algorithm, data, signature, context, key);
+            case EC:
+            case EC_HSM:
+                return ecKeyCryptographyClient.verifyDataAsync(algorithm, data, signature, context, key);
+            default:
+                return Mono.error(new UnsupportedOperationException(String.format("Encrypt Async is not supported for Key Type: %s", key.kty().toString())));
+        }
+    }
+
+    private void unpackAndValidateId(String keyId) {
+        if (keyId != null && keyId.length() > 0) {
+            try {
+                URL url = new URL(keyId);
+                String[] tokens = url.getPath().split("/");
+                String endpoint = url.getProtocol() + "://" + url.getHost();
+                String keyName = (tokens.length >= 3 ? tokens[2] : null);
+                version = (tokens.length >= 4 ? tokens[3] : null);
+                if(Strings.isNullOrEmpty(endpoint)) {
+                    throw new IllegalArgumentException("Key endpoint in key id is invalid");
+                } else if (Strings.isNullOrEmpty(keyName)) {
+                    throw new IllegalArgumentException("Key name in key id is invalid");
+                } else if(Strings.isNullOrEmpty(version)) {
+                    throw new IllegalArgumentException("Key version in key id is invalid");
+                }
+            } catch (MalformedURLException e) {
+                e.printStackTrace();
```

Comment:
Don't use `printStackTrace()`. Wrap it in `IllegalArgumentException` and throw or if you don't want to throw, just log warning.

Reply:
Updated to throw `IllegalArgumentException`.